lib/src/components/NxTag/__tests__/NxTag.test.tsx

```diff
@@ -4,124 +4,69 @@
 
-  it('renders NxTag with the `nx-tag` class', function() {
-    expect(getShallowComponent().children().find('.nx-tag')).toExist();
-  });
+describe('NxTag', function() {
+  type PropsWithRef = PublicProps & RefAttributes<HTMLLabelElement>;
+  const renderEl = rtlRenderElement<PropsWithRef>(NxTag, { children: 'basic tag '});
 
   it('forwards a ref', function() {
-    const ref = React.createRef<HTMLLabelElement>();
-    const component = mount(<><NxTag ref={ref}>Hello</NxTag></>);
-    expect(component.find('.nx-tag').getDOMNode()).toBe(ref.current);
-  });
-
-  it('correctly assigns supplied class', function() {
-    const classComponent = getShallowComponent({ className: 'foo' });
-    expect(classComponent.find('.nx-tag')).toHaveClassName('foo');
-  });
+    const ref = React.createRef<HTMLLabelElement>(),
+        renderedEl = renderEl({ ref });
 
-  it('correctly assigns supplied id', function() {
-    const idComponent = getShallowComponent({ id: 'test-id' });
-    expect(idComponent.find('.nx-tag')).toHaveProp('id', 'test-id');
+    expect(ref.current).toBe(renderedEl);
   });
 
-  it('renders the supplied text inside .nx-tag__text', function() {
-    const tagChildren = getShallowComponent({ children: 'tag text' });
-    expect(tagChildren.find('.nx-tag__text')).toHaveText('tag text');
+  it('correctly assigns supplied class', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
 
-    expect(tagChildren.children().first()).toMatchSelector('label');
-    // expect(tagChildren.children().at(1)).toMatchSelector('span');
-  });
+    expect(customEl).toHaveClass('foo');
 
-  it('sets the nx-selectable-color--indigo class if no color prop is passed', function() {
-    expect(getShallowComponent().find('.nx-selectable-color--indigo')).toExist();
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
-  it('sets the color class using the color if it is provided', function() {
-    const colorComponent = getShallowComponent({ color: 'orange' });
-    expect(colorComponent.find('.nx-selectable-color--orange')).toExist();
+  it('correctly assigns supplied id', function() {
+    const el = renderEl({ id: 'test-id' });
+    expect(el).toHaveAttribute('id', 'test-id');
   });
 
-  it('checks the children appear in the correct order, text then the icon', function() {
-    const getSelectedComponent = getShallowComponent({
-      children: 'foo bar',
-      selectedIcons: <NxFontAwesomeIcon icon={faPlusCircle} className="nx-tag__action" />
-    });
-
-    expect(getSelectedComponent.find('.nx-tag').children().first()).toMatchSelector('span');
-    expect(getSelectedComponent.find('.nx-tag').children().last()).toHaveClassName('nx-tag__action');
+  it('renders the supplied text', function() {
+    const el = renderEl({ children: 'tag text' });
+    expect(el).toHaveTextContent('tag text');
   });
 });
 
 describe('NxSelectableTag', function() {
```

Comment:
`NxSelectableTag`s include a checkbox with the `switch` role and who's `checked` attr reflects the `selected` prop.  That should all be tested

Reply:
> `NxSelectableTag`s include a checkbox with the `switch` role and who's `checked` attr reflects the `selected` prop. The checkbox's a11y name is equal to the tag text. That should all be tested

Added here: ad6e6ee522d01288b3b2dbc9154bb640228bb787